### PR TITLE
Refactor `<HashtagHeader>` to TypeScript

### PR DESCRIPTION
**Pull request by Gargron**

### CODE REVIEW

File: app/javascript/mastodon/api/tags.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { apiRequestPost, apiRequestGet } from 'mastodon/api';
+import type { ApiHashtagJSON } from 'mastodon/api_types/tags';
+
+export const apiGet = (tagId: string) =>
```
Comment (ClearlyClaire) [inline app/javascript/mastodon/api/tags.ts:4]: Wouldn't this be clearer as `apiGetTag` or `apiFetchTag`? I do realize that we have some inconsistencies in naming across modules with `Get` VS `Fetch` and omitting or mentioning the class name (I think the only existing cases where the names are really lacking are `apiCreate` and `apiUpdate` in `app/javascript/mastodon/api/lists.ts`).

File: app/javascript/mastodon/features/hashtag_timeline/components/hashtag_header.tsx [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+    if (tag.following) {
+      setTag((hashtag) => hashtag && { ...hashtag, following: false });
+
+      void dispatch(unfollowHashtag({ tagId })).then((result) => {
+        if (isFulfilled(result)) {
+          setTag(result.payload);
+        }
+
+        return '';
```
Comment (ClearlyClaire) [inline app/javascript/mastodon/features/hashtag_timeline/components/hashtag_header.tsx:121]: That looks a bit weird. Why return an empty string?

File: app/javascript/mastodon/features/hashtag_timeline/components/hashtag_header.tsx [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+      });
+    } else {
+      setTag((hashtag) => hashtag && { ...hashtag, following: true });
+
+      void dispatch(followHashtag({ tagId })).then((result) => {
+        if (isFulfilled(result)) {
+          setTag(result.payload);
+        }
+
+        return '';
```
Comment (ClearlyClaire) [inline app/javascript/mastodon/features/hashtag_timeline/components/hashtag_header.tsx:131]: Ditto.

File: app/javascript/mastodon/features/hashtag_timeline/components/hashtag_header.tsx (hunk 2 above)
Comment (Gargron) [reply app/javascript/mastodon/features/hashtag_timeline/components/hashtag_header.tsx:121]: It's an [eslint rule](https://github.com/eslint-community/eslint-plugin-promise/blob/main/docs/rules/always-return.md). I don't like it either! But it's not harmful. I guess it makes it clearer you can't chain another `then` on.

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**